### DISCUSSION TITLE
Would supabase be interested in a PR to add timescaledb toolkit

### DISCUSSION THREAD

**Discussion by mwudka** [supabase/supabase#17108] [Contribute to Supabase]
Howdy!

When trying out supabase, I found that supabase includes timescaledb. Sadly, it does not include the closely related timescaledb-toolkit extension, which includes many functions needed to make timescaledb useful. Would supabase be interested in a PR to add this toolkit?

I hacked a rough prototype, and learned that the changes/work are roughly:
1. Confirm licensing status. I believe the toolkit is under the same dual licensing structure as timescaledb itself, but I will need to double-check this
2. Add another section to the Dockerfile that downloads and installs the timescaledb-toolkit package, following the pattern used elsewhere in the Dockerfile to install other extensions via deb package.
3. Extend the ansible playbook task for postgres-extensions to also install the timescaledb-toolkit package
4. Write SQL migration scripts for both that create the toolkit extension (this is needed because the toolkit is a C extension, which requires a superuser to activate)
5. Test the docker, AWS arm64, and DO images to ensure they still boot cleanly and that timescaledb works properly.

Also, I noticed that timescaledb is built with the non-Apache features disabled. Would it be OK to change that to a flag that defaults to maintaining the current behavior? That way, nothing would change by default but it would be easy for folks self-hosting to run the community version instead.

Thanks,
Manuel

**Comment by encima** [comment]
As the [Timescale License](https://github.com/timescale/timescaledb/blob/main/tsl/LICENSE-TIMESCALE) prevents the use for Database-as-a-service. Supabase providing this, and maintaining it, would be a bit of a minefield. For those self-hosting, you are welcome to create post-install scripts or similar but we cannot officially support this.

Note also that the toolkit extension does not have builds for ARM so it would need to be built yourself if that is your architecture

**Comment by tonven** [comment]
@mwudka I am self-hosting supabase and trying to add timescaledb and timescaledb-toolkit extension. Do you have an example of script that can be used in docker compose and / or during local development to enable those extenstions? Thank you!

**Reply by paulwer** [reply]
@mwudka did you found a solution?

**Comment by servrox** [comment]
Wrote a thread about it, maybe this helps. Came up to this issue while searching for solutions.
https://x.com/servrox1337/status/2023679230530211958
